### ISSUE TITLE
vkCreateImage fails for RGB images on GTX960M with 384.11 drivers

### ISSUE THREAD

**Issue by skalarproduktraum** [scenerygraphics/scenery#151]
As reported by @maarzt
[console_output.txt](https://github.com/scenerygraphics/scenery/files/1952751/console_output.txt)
[hs_err_pid16232.log](https://github.com/scenerygraphics/scenery/files/1952752/hs_err_pid16232.log)

Relevant part of the console output:
```
SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - Recording scene command buffer graphics.scenery.backends.vulkan.VulkanCommandBuffer@39c9ff98 for pass DeferredLighting...
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - Recording scene command buffer graphics.scenery.backends.vulkan.VulkanCommandBuffer@2a1fc259 for pass ForwardShading...
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - HDR: descriptor sets are ShaderParameters-HDR, input-HDR-0
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - pipeline provides ShaderParameters, Inputs-0
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - Ubo position: 0
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - Adding DS#0 for input-HDR-0 to required pipeline DSs
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - Adding DS#1 for ShaderParameters-HDR to required pipeline DSs
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - HDR: Requires 0 dynamic offsets
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - FXAA: descriptor sets are ShaderParameters-FXAA, input-FXAA-0
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - pipeline provides ShaderParameters, Inputs-0
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - Ubo position: 0
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - Adding DS#0 for input-FXAA-0 to required pipeline DSs
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - Adding DS#1 for ShaderParameters-FXAA to required pipeline DSs
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - FXAA: Requires 0 dynamic offsets
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - Force command buffer re-recording, as geometry for box has been updated
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - Reusing existing vertex+index buffer for box update
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - Using VulkanBuffer 0x7F9FD5ED7AA0 for vertex+index storage
[SciJava-52cb1faf-Thread-1] DEBUG VulkanRenderer - box will have diffuse texture from fromBuffer:diffuse in slot 1
[SciJava-52cb1faf-Thread-1] ERROR VulkanRenderer - Call to create staging image failed: A validation layer found an error.
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f9fcf3776f0, pid=16232, tid=0x00007f9ff8806700
#
# JRE version: Java(TM) SE Runtime Environment (8.0_161-b12) (build 1.8.0_161-b12)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.161-b12 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libnvidia-glcore.so.384.111+0x14386f0]
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/arzt/devel/scenery/hs_err_pid16232.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.

```

**Comment by skalarproduktraum** [comment]
@maarzt, we're gonna need more flags! Could you run this with `-Dscenery.VulkanRenderer.EnableValidations=true` please and post the output?

**Comment by maarzt** [comment]
I added the flag, that's the new output:
[console_output_validation.txt](https://github.com/scenerygraphics/scenery/files/1953109/console_output_validation.txt)

**Comment by maarzt** [comment]
BTW, when I run `sc.iview.Main` class with the Vulkan renderer, then I get this result:
![screenshot_scenery_main_linetest](https://user-images.githubusercontent.com/24407711/39334298-edaa3772-4962-11e8-8ac5-781061e6e8d0.png)
Crazy colors, it look much different than the OpenGL version.
And when I run `meshTest` in `sc.iview.Main` with Vulkan, then it crashes reliably, and produces the core dump and output I previously posted.

**Comment by skalarproduktraum** [comment]
Hey @maarzt! Thank you for continued reporting, I've found the issue and tentatively fixed it. 
Turns out, R8G8B8 images are not very widely supported (see e.g. http://vulkan.gpuinfo.org/listreports.php?linearformat=R8G8B8_SNORM&option=not). If images are read from a file, they get converted to RGBA, this is however not the case for images read from a buffer.

I've fixed this issue in the `vulkan-pad-rgb-textures` branch. Could you check that one out, run `mvn test -Dtest=ProceduralTextureExample` and see whether this fixes the issue for you as well? Thanks! :+1:

**Comment by skalarproduktraum** [comment]
And to your last comment, the lines are actually supposed to be there as they are when running `meshTest`. Them not being there was related to a bug in the UBO serialisation, #152, which is now fixed.

**Comment by skalarproduktraum** [comment]
Resolved in 0.3.1, but more current drivers are required. Version 390.192 works.
